File: UI/src/app/models/response/responseBrowse.ts
// Browse-grid response payloads — mirrors UI/src-tauri/src/models/response/response_browse.rs.

// One entry (folder or image) in the Browse view.
export interface BrowseEntry {
  name: string;
  path: string;
}

export interface responseBrowseDirectory {
  current: string;
  breadcrumb: BrowseEntry[];
  folders: BrowseEntry[];
  images: BrowseEntry[];
}

export interface responseGetThumbnail {
  path: string;
}

File: UI/src-tauri/src/models/response/response_browse.rs
// Browse-grid response payloads.
use serde::Serialize;

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct BrowseEntry {
    pub name: String,
    pub path: String,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ResponseBrowseDirectory {
    pub current: String,
    pub breadcrumb: Vec<BrowseEntry>,
    pub folders: Vec<BrowseEntry>,
    pub images: Vec<BrowseEntry>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct ResponseGetThumbnail {
    pub path: String,
}
